tests/CMakeLists.txt

```diff
@@ -1,15 +1,11 @@
-function(add_cmake_test_test)
-    foreach(testi ${ARGN})
-        set(full_path ${CMAKE_CURRENT_LIST_DIR}/${testi}.cmake)
-        add_test(
-            NAME "${testi}"
-            COMMAND ${CMAKE_COMMAND}
-            -DCMAKE_MODULE_PATH=${PROJECT_SOURCE_DIR}
-            -P ${full_path}
-        )
-    endforeach()
-endfunction()
-
-add_subdirectory(asserts)
-add_subdirectory(detail_)
-add_subdirectory(tutorials)
+cmake_minimum_required(VERSION "${_ct_min_cmake_version}") #Required for FetchContent_MakeAvailable()
+
+project(CMakeTest LANGUAGES C) #Need C language so CMake doesn't complain about no linker language for some empty files used for tests
```

Comment:
If you include another `project` command I think CMake will complain. You can use [`enable_language`](https://cmake.org/cmake/help/latest/command/enable_language.html#:~:text=enable_language%20%28%3Clang%3E%20%5BOPTIONAL%5D%29%20Enables%20support%20for%20the%20named,C%2C%20CUDA%2C%20OBJC%2C%20OBJCXX%2C%20Fortran%2C%20ISPC%2C%20and%20ASM.) to turn on a new language

Reply:
Whoops, I didn't catch that cause I was using add_subdirectory() instead of include(). It's been removed now, and languages don't need to be set there anymore anyways since we moved test execution to the testing stage.